py/selenium/webdriver/common/service.py

```diff
@@ -89,33 +90,21 @@ def start(self) -> None:
         try:
             cmd = [self.path]
             cmd.extend(self.command_line_args())
-            self.process = subprocess.Popen(
-                cmd,
-                env=self.env,
-                close_fds=system() != "Windows",
-                stdout=self.log_file,
-                stderr=self.log_file,
-                stdin=PIPE,
-                creationflags=self.creation_flags,
-            )
-            log.debug(f"Started executable: `{self.path}` in a child process with pid: {self.process.pid}")
-        except TypeError:
-            raise
-        except OSError as err:
-            if err.errno == errno.ENOENT:
-                raise WebDriverException(
-                    f"'{os.path.basename(self.path)}' executable needs to be in PATH. {self.start_error_message}"
-                )
-            elif err.errno == errno.EACCES:
-                raise WebDriverException(
-                    f"'{os.path.basename(self.path)}' executable may have wrong permissions. {self.start_error_message}"
-                )
-            else:
-                raise
-        except Exception as e:
-            raise WebDriverException(
-                f"The executable {os.path.basename(self.path)} needs to be available in the path. {self.start_error_message}\n{str(e)}"
-            )
+            self._start_process(cmd)
+        except WebDriverException as err:
+            if "executable needs to be in PATH" in err.msg:
+                logger.info("driver not found in PATH, trying Selenium Manager")
+                browser = self.__class__.__module__.split(".")[-2]
```

Comment:
what browsers does this support; is this the most reliable way to grab the browser?

Reply:
Chrome, Firefox & Edge (so everything but IE); but good call, should make sure it is one of those. 
We could alternately add a constant to each of the subclasses, but I figured the module was unlikely to change.
This is different from other languages because it gets the driver differently, but now that I'm thinking of it, maybe we don't want to use Selenium manager if someone is explicitly asking for a path that is invalid...